core/session.ts

```diff
@@ -216,7 +217,8 @@ export class Session {
     }
   }
 
-  public resolve(ref: Resolvable): string {
+  public resolve(ref: Resolvable | string[], ...rest: string[]): string {
```

Comment:
Shouldn't the string here not be pluralised?
```suggestion
  public resolve(ref: Resolvable | string, ...rest: string[]): string {
```

Reply:
hmm, no.

we need to accept any of the following:
1) a `Resolvable`, i.e. a single string corresponding to the target's `name`; OR the full `Target`
2) an explicit array, i.e. `["db", "schema", "name"]`
3) that array but without the square brackets, i.e. using the `rest` parameter